chromadb/api/__init__.py

```diff
@@ -412,6 +413,19 @@ def max_batch_size(self) -> int:
         to submit_embeddings."""
         pass
 
+    @abstractmethod
+    def env(
+        self,
+        system_info_flags: Optional[SystemInfoFlags] = None,
+    ) -> Dict[str, Any]:
```

Comment:
Can we strongly type the output here with a TypedDict

Reply:
Done. I've also flatten the structure so we don't have nested stuff:

```py
class SystemInfo(TypedDict):
    chroma_version: str
    python_version: str
    is_persistent: bool
    api: str
    datetime: str
    os: str
    os_version: str
    os_release: Optional[str]
    cpu_architecture: str
    cpu_count: Optional[int]
    cpu_usage: float
    memory_free: int
    memory_total: int
    process_memory_rss: int
    process_memory_vms: int
    persistent_disk_total: Optional[int]
    persistent_disk_used: Optional[int]
    persistent_disk_free: Optional[int]
    mode: ChromaMode
  ```